tdrs-backend/tdpservice/parsers/management/commands/seed_db.py

```diff
@@ -0,0 +1,200 @@
+            user=User.objects.get_or_create(username='system')[0],
+            stt=stt,
+            year=year,
+            quarter=quarter,
+            original_filename=original_filename,
+            section=section,
+            version=random.randint(1, 1993415),
+        )
+
+
+        d.file.save(file_name, ContentFile(file_data))
+    except django.db.utils.IntegrityError as e:
+        pass
+    return d
+
+
+def validValues(schemaMgr, field, year):
+    '''Takes in a field and returns a line of valid values.'''
+
+    field_len = field.endIndex - field.startIndex
+
+    if field.name == 'RecordType':
+        return schemaMgr.record_type
+    if field.name == 'SSN':
+        # only used by recordtypes 2,3,5 
+        # TODO: reverse the TransformField logic to 'encrypt' a random number
+        field_format = '?' * field_len
+    elif field.name in ('RPT_MONTH_YEAR', 'CALENDAR_QUARTER'):
+        lower = 1 # TODO: get quarter and use acceptable range for month
+        upper = 12
+
+        month = '{}'.format(random.randint(lower, upper)).zfill(2)
+        field_format = '{}{}'.format(year, str(month))
+    else:
+        field_format = '#' * field_len
+    return fake.bothify(text=field_format)
+
+
+def make_line(schemaMgr, section, year):
+    '''Takes in a schema manager and returns a line of data.'''
+    line = ''
+
+    for row_schema in schemaMgr.schemas:  # this is to handle multi-schema like T6
+        for field in row_schema.fields:
+            line += validValues(row_schema, field, year)
+    return line + '\n'
+
+def make_HT(schemaMgr, prog_type, section, year, quarter, stt):
+    line = ''
+
+    if type(schemaMgr) is RowSchema:
+        if schemaMgr.record_type == 'HEADER':
+            # e.g. HEADER20201CAL000TAN1ED
+            for field in schemaMgr.fields:
+                if field.name == 'title':
+                    line += 'HEADER'
+                elif field.name == 'year':
+                    line += '{}'.format(year)
+                elif field.name == 'quarter':
+                    line += quarter[1:]  # remove the 'Q', e.g., 'Q1' -> '1'
+                elif field.name == 'type':
+                    line += section
+                elif field.name == 'state_fips':
+                    if stt.state is not None:  # this is a tribe
+                        my_stt = stt.state
+                    else:
+                        my_stt = stt
+                    line += '{}'.format(my_stt.stt_code).zfill(2)
+                elif field.name == 'tribe_code':
+                    if stt.type == 'tribe':
+                        line += stt.stt_code
+                    else:
+                        line += '000'
+                elif field.name == 'program_type':
+                    line += prog_type
+                elif field.name == 'edit':
+                    line += '1'
+                elif field.name == 'encryption':
+                    line += 'E'
+                elif field.name == 'update':
```

Comment:
I really wish I could condense this, but the value-add of this script isn't to generate a bunch of files that fail at the first line and get auto-rejected. Having a good header is paramount but there are enough differences in the headers of various files that I can no longer just use the hardcoded one-liner I had previously.